Source/WebKit/UIProcess/API/Cocoa/WKDownloadDelegatePrivate.h

```diff
@@ -50,9 +50,25 @@ WK_SWIFT_UI_ACTOR
  @param download The download for which we need a placeholder policy
  @param completionHandler The completion handler that should be invoked with the chosen policy
  @discussion The placeholder policy specifies whether a placeholder file should be created in
- the Downloads directory when the download is in progress.
+ the Downloads directory when the download is in progress. If the client opts out of the
+ placeholder feature, it can choose to provide a custom URL to publish progress against.
+ This is useful if the client maintains it's own placeholder file.
  */
-- (void)_download:(WKDownload *)download decidePlaceholderPolicy:(void (^)(_WKPlaceholderPolicy))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+- (void)_download:(WKDownload *)download decidePlaceholderPolicy:(void (^)(_WKPlaceholderPolicy, NSURL *))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
```

Comment:
Can we replace _WKPlaceholderPolicy with the nullness of the URL?

Reply:
That's a good point! I don't believe we can, since we need to be able to distinguish between the use cases where the client does not want a placeholder at all, and the use case where the client is maintaining the placeholder and does not want the placeholder provided by the system.

Thanks for reviewing! I will address the comments in the upcoming patch.